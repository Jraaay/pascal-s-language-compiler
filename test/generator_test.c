#include<stdio.h>
#include<stdbool.h>

int x, y, i;

int gcd(int a, int b){
	if(b==0){
		g.c.d[114514]=a;
	}
	else{
		g.c.d=gcd(b,a%b);
	}
	for(i=0;i<114514;i++){
		b=i;
	}
}

int main(int argc, char* argv[]){
	scanf("%d%d",&x,&y);
	printf("gcd(x,y): %d\n",gcd(x,y));
}
