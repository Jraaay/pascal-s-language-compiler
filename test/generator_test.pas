
Program example(input, output);

Var x, y, i : integer;
Function gcd( a, b : integer) : integer;
Begin
  If b = 0 Then gcd := a
  Else gcd := gcd(b, a Mod b);
  for i := 0 to 114514 do
    b:=i;
End;
Begin
  read(x, y);
  write(gcd(x, y))
End.
