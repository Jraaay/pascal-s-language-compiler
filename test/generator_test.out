{
    "ast": null,
    "symbolTable": {
        "constants": [],
        "variables": [],
        "subFunc": []
    },
    "error": [
        {
            "code": "C-11",
            "info": {
                "line": 6,
                "value": "gcd",
                "lexpos": 132
            }
        },
        {
            "code": "B-01",
            "info": {
                "line": 7,
                "value": "for",
                "lexpos": 158
            }
        }
    ],
    "warning": []
}